src/main/resources/api/openapi.yaml

```diff
@@ -194,66 +227,86 @@ components:
           type: integer
           format: int64
           example: 1514194867
+        connectionType:
+          type: string
+          enum:
+            - Incoming
+            - Outgoing
 
     NodeInfo:
       type: object
       required:
         - name
-        - version
+        - appVersion
         - fullHeight
         - headersHeight
         - bestFullHeaderId
+        - previousFullHeaderId
+        - bestHeaderId
         - stateRoot
         - stateType
         - stateVersion
-        - previousFullHeaderId
         - isMining
         - votes
         - peersCount
         - unconfirmedCount
         - difficulty
+        - currentTime
+        - launchTime
       properties:
         name:
           type: string
           example: my-node-1
-        version:
+        appVersion:
           type: string
           example: 0.0.1
         fullHeight:
           type: integer
           format: int32
-          description: Can be 0 if state is empty
+          description: Can be 'null' if state is empty (no full block is applied since node launch)
           minimum: 0
           example: 667
+          nullable: true
         headersHeight:
           type: integer
           format: int32
-          description: Can be 0 if state is empty
+          description: Can be 'null' if state is empty (no header applied since node launch)
           minimum: 0
           example: 667
+          nullable: true
         bestFullHeaderId:
-          $ref: '#/components/schemas/ModifierId'
+          type: string
+          description: Can be 'null' if no full block is applied since node launch
+          nullable: true
+          allOf:
+            - $ref: '#/components/schemas/ModifierId'
```

Comment:
I guess we can introduce a new type like `ModifierIdOption`, which is nullable, and use this type instead of referencing `allOf` ModifierId in all these places

Reply:
Not sure this is good for this particular case, because we can not use any sibling elements with $ref. See https://swagger.io/docs/specification/using-ref/

So, if we need unique descriptions here, we need to repeat property definitions too.